pydra/engine/core.py

```diff
@@ -629,26 +737,22 @@ def _extract_input_el(self, inputs, inp_nm, ind):
 
     def get_input_el(self, ind):
         """Collect all inputs required to run the node (for specific state element)."""
-        if ind is not None:
-            # TODO: doesn't work properly for more cmplicated wf (check if still an issue)
-            state_dict = self.state.states_val[ind]
-            input_ind = self.state.inputs_ind[ind]
-            inputs_dict = {}
-            for inp in set(self.input_names):
-                if f"{self.name}.{inp}" in input_ind:
-                    inputs_dict[inp] = self._extract_input_el(
-                        inputs=self.inputs,
-                        inp_nm=inp,
-                        ind=input_ind[f"{self.name}.{inp}"],
-                    )
-                else:
-                    inputs_dict[inp] = getattr(self.inputs, inp)
-            return state_dict, inputs_dict
-        else:
-            # todo it never gets here
-            breakpoint()
-            inputs_dict = {inp: getattr(self.inputs, inp) for inp in self.input_names}
-            return None, inputs_dict
+        # TODO: doesn't work properly for more cmplicated wf (check if still an issue)
+        input_ind = self.state.inputs_ind[ind]
+        inputs_dict = {}
+        for inp in set(self.input_names):
+            if f"{self.name}.{inp}" in input_ind:
+                inputs_dict[inp] = self._extract_input_el(
```

Comment:
Follow-up to my comment above, now we've got `inp`, `inpt`, `input` and `inputs`. Then `inp_nm`, `input_ind`, ...

It's getting very difficult for me to keep track of variable meaning in my limited short-term memory.

Reply:
I typically use inpt to differentiate from the inbuilt function